### PR TITLE
feat: Enable/Disable logging in tests via RUST_LOG environment variable

### PR DESCRIPTION
**Pull request by alamb**
TLDR; This PR allows runtime enabling/disable of logging in tests

# Rationale

I am trying to be more productive.

My debugging flow in the query engine often looks like
1. run tests
2. find those that fail
3. add `test_util::enable_logging()`
4. rerun to get logging

I am trying to avoid the recompile/rerun steps 3 and 4

# Changes:

Add a function to enable logging if `RUST_LOG` is set in the environment, and does nothing if `RUST_LOG` is not set

So now, you can enable logs in the tests without a recompile via:

```
RUST_LOG=debug cargo test -p query
```

# Examples:

## Tests as Normal
```
cargo test -p server -- sql
```

Which runs the tests like you would hope, nice and quiet like

```
cd /Users/alamb/Software/influxdb_iox2 && cargo test -p server -- sql
    Blocking waiting for file lock on build directory
   Compiling server v0.1.0 (/Users/alamb/Software/influxdb_iox2/server)
    Finished test [unoptimized + debuginfo] target(s) in 23.50s
     Running target/debug/deps/server-c6c98a959c339054

running 6 tests
test query_tests::sql::sql_select_from_disk ... ok
test query_tests::sql::sql_select_from_cpu_with_projection ... ok
test query_tests::sql::sql_select_from_cpu ... ok
test query_tests::sql::sql_select_from_cpu_pred ... ok
test query_tests::sql::sql_select_from_cpu_with_projection_and_pred ... ok
test query_tests::sql::sql_select_from_cpu_group ... ok

test result: ok. 6 passed; 0 failed; 0 ignored; 0 measured; 46 filtered out

   Doc-tests server

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out


Compilation finished at Fri Feb 12 06:56:32
```
## Log Connoisseur Mode
If you want a heaping helping of logs, you can do:

```
RUST_LOG=debug cargo test -p server -- sql
```

Which produces
```
cd /Users/alamb/Software/influxdb_iox2 && RUST_LOG=debug cargo test -p server -- sql
    Finished test [unoptimized + debuginfo] target(s) in 0.36s
     Running target/debug/deps/server-c6c98a959c339054

running 6 tests
[2021-02-12T11:57:23Z DEBUG sqlparser::parser] Parsing sql 'SELECT user, region from cpu'...
[2021-02-12T11:57:23Z DEBUG sqlparser::parser] Parsing sql 'SELECT count(*) from cpu group by region'...
[2021-02-12T11:57:23Z DEBUG sqlparser::parser] parsing expr
[2021-02-12T11:57:23Z DEBUG sqlparser::parser] Parsing sql 'SELECT user, region from cpu where time > 125'...
[2021-02-12T11:57:23Z DEBUG sqlparser::parser] Parsing sql 'SELECT * from disk'...
[2021-02-12T11:57:23Z DEBUG sqlparser::parser] parsing expr
... (SNIP) ...
    ----
[2021-02-12T11:57:27Z DEBUG datafusion::physical_plan::coalesce_batches] Combined 1 batches containing 1 rows
test query_tests::sql::sql_select_from_cpu_with_projection_and_pred ... ok

test result: ok. 6 passed; 0 failed; 0 ignored; 0 measured; 46 filtered out

   Doc-tests server

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out


Compilation finished at Fri Feb 12 06:57:27
```

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

**Comment by e-dard** [comment]
I'm a little confused why `RUST_LOG` doesn't just work in tests as it is?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -99,6 +99,25 @@ The `cargo` build tool runs tests as well. Run:
...
+
+To enable logging to stderr during a run of `cargo test` set the Rust
+`RUST_LOG` environment varable. For example, to see all INFO messages:
+
+```shell
+RUST_LOG=info cargo test --workspace
+```
+
+Since this feature uses
+[`env_logger`](https://crates.io/crates/env_logger) internally, you
```
Comment (tustvold) [inline CONTRIBUTING.md:112]: I could be wrong about this, but isn't it actually [EnvFilter](https://docs.rs/tracing-subscriber/0.2.15/tracing_subscriber/filter/struct.EnvFilter.html) which is more expressive?

Comment (tustvold) [reply CONTRIBUTING.md:112]: Oh nvm, we use envlogger for the tests and tokio-tracing for runtime - this is perhaps not ideal?

Comment (alamb) [reply CONTRIBUTING.md:112]: That is a good point -- I will update this PR to use the same logging system we use in the server

**Comment by alamb** [comment]
> I'm a little confused why RUST_LOG doesn't just work in tests as it is?

@e-dard  I think it is because nothing ever "sets up" the logging system in the test harness

I also double checked that RUST_LOG doesn't do anything on main:

```
cd /Users/alamb/Software/influxdb_iox2 && RUST_LOG=debug cargo test -p server -- sql
   Compiling test_helpers v0.1.0 (/Users/alamb/Software/influxdb_iox2/test_helpers)
   Compiling data_types v0.1.0 (/Users/alamb/Software/influxdb_iox2/data_types)
   Compiling packers v0.1.0 (/Users/alamb/Software/influxdb_iox2/packers)
   Compiling query v0.1.0 (/Users/alamb/Software/influxdb_iox2/query)
   Compiling read_buffer v0.1.0 (/Users/alamb/Software/influxdb_iox2/read_buffer)
   Compiling mutable_buffer v0.1.0 (/Users/alamb/Software/influxdb_iox2/mutable_buffer)
   Compiling server v0.1.0 (/Users/alamb/Software/influxdb_iox2/server)
    Finished test [unoptimized + debuginfo] target(s) in 21.97s
     Running target/debug/deps/server-a765ef0b1a53c911

running 8 tests
test query_tests::sql::sql_select_from_disk ... ok
test query_tests::sql::sql_select_from_cpu ... ok
test query_tests::sql::sql_select_from_cpu_with_projection ... ok
test query_tests::sql::sql_select_from_cpu_pred ... ok
test query_tests::sql::sql_select_from_cpu_with_projection_and_pred ... ok
test query_tests::sql::sql_select_with_schema_merge_subset ... ok
test query_tests::sql::sql_select_with_schema_merge ... ok
test query_tests::sql::sql_select_from_cpu_group ... ok

test result: ok. 8 passed; 0 failed; 0 ignored; 0 measured; 46 filtered out

   Doc-tests server

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out


Compilation finished at Fri Feb 12 07:36:11
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (alamb) [reply CONTRIBUTING.md]: Switched to use EnvFilter in https://github.com/influxdata/influxdb_iox/pull/793/commits/ef991dd19d22d0bc41a0327578caf70aadd6a79a

**Review by tustvold — APPROVED** [review]
